src/server/field.rs

```diff
@@ -138,63 +125,42 @@ pub struct ContentDisp {
 }
 
 impl ContentDisp {
-    fn parse(headers: &[StrHeader]) -> Option<ContentDisp> {
+    fn parse(headers: &[StrHeader]) -> Result<ContentDisp, ParseHeaderError> {
         if headers.is_empty() {
-            return None;
+            return Err(ParseHeaderError::Missing);
```

Comment:
This isn't an error, there's just no more fields. A missing error would be if `headers` wasn't empty, but `Content-Disposition` (required header) wasn't found.